frontend/src/pages/accounts/profile.test.tsx

```diff
@@ -570,6 +571,136 @@ describe("The dashboard", () => {
     mockConfigModule.getRuntimeConfig.mockReturnValue(mockedConfig);
   });
 
+  it("shows the category filter button if the user has Premium", () => {
+    setMockProfileDataOnce({ has_premium: true });
+    setMockAliasesDataOnce({ random: [{ enabled: true, id: 42 }], custom: [] });
+
+    render(<Profile />);
+
+    const categoryFilterButton = screen.getByRole("button", {
+      name: "l10n string: [profile-filter-category-button-tooltip], with vars: {}",
+    });
+
+    expect(categoryFilterButton).toBeInTheDocument();
+  });
+
+  it("has a category filter for the list of aliases that can be closed and applied, which is available to users with premium", () => {
+    setMockProfileDataOnce({ has_premium: true });
+    setMockAliasesDataOnce({
+      random: [
+        getMockRandomAlias({ address: "address1" }),
+        getMockRandomAlias({ address: "address2" }),
+      ],
+      custom: [getMockCustomAlias({ address: "address3" })],
+    });
+
+    render(<Profile />);
+
+    // Open and select a category filter option
+    const categoryFilterButton = screen.getByRole("button", {
+      name: "l10n string: [profile-filter-category-button-tooltip], with vars: {}",
+    });
+    userEvent.click(categoryFilterButton);
+
+    const categoryFilterCheckboxCustomMask = screen.getByLabelText(
+      "l10n string: [profile-filter-category-option-custom-masks], with vars: {}"
+    );
+    userEvent.click(categoryFilterCheckboxCustomMask);
+
+    // Apply category filter selection
+    const categoryFilterApplyButton = screen.getByRole("button", {
+      name: "l10n string: [profile-label-apply], with vars: {}",
+    });
+    userEvent.click(categoryFilterApplyButton);
+
+    // The blocked label serves as a proxy for an alias card
+    const aliasCardBlockedLabels = screen.queryByText(
+      "l10n string: [profile-label-blocked], with vars: {}"
+    );
+
+    expect(aliasCardBlockedLabels).toBeInTheDocument();
```

Comment:
While this does verify whether aliases that are set to blocking are still being displayed, it doesn't test whether the rest of the aliases are actually hidden.

Part of the challenge is that there's an error in the mocks, which I submitted a fix for in https://github.com/mozilla/fx-private-relay/pull/1879. With that fix, you should be able to do the following, not needing the "blocked" label as a proxy:

```suggestion
    const randomAlias1 = screen.queryByText(/address1/);
    const randomAlias2 = screen.queryByText(/address2/);
    expect(randomAlias1).not.toBeInTheDocument();
    expect(randomAlias2).not.toBeInTheDocument();
    const customAlias = screen.queryByText(/address3/);
    expect(customAlias).toBeInTheDocument();
```

Something similar could apply to the tests following this one.

Reply:
Thanks for the suggestion and the fix! I didn’t really realise that the mask addresses weren’t rendered. I applied your suggestion to the other tests as well.

Reply:
Awesome, looking sharp! Feel free to merge :)